Added logic to handle relative path URLs in <link> elements

rod-o-walker opened the pull request:
In working with this module, I noticed some issues with CSS stylesheets not being applied properly in the child windows. I isolated the issue to ones with relative paths, which will not work, because the child window has no URL of its own. So I added some logic to prepend the host and path of the main window to any relative URLs.
Thx.

Shemesh commented:
hi @rod-o-walker, myself also noticed this issue.
i'm working on some major changes, will submit and publish soon.
here is how i've fixed this issue of the link stylesheet.
```
document.head.querySelectorAll('link[rel="stylesheet"][type="text/css"]').forEach(node => {
  this.popoutWindow.document.head.insertAdjacentHTML('beforeend',
  `<link rel="stylesheet" type="text/css" href="${(node as HTMLLinkElement).href}">`);
});
```

what do you think of my solution?

rod-o-walker commented:
That is a cleaner selection strategy. However, it does not address the
issue with relative URLs that I identified.

You will notice in the index.html of the test app, I added following

<link rel="stylesheet" href="/assets/test.css">

With the fix I added, when the child window is opened and the link tag is
copied over, it will update the relative path to an absolute one.

<link rel="stylesheet" href="http://localhost:4200/assets/test.css">

Thx.
______________________________________________

Rod Walker
e-mail: rod.o.walker@gmail.com


On Tue, Dec 15, 2020 at 3:33 AM Shemesh <notifications@github.com> wrote:

> hi @rod-o-walker <https://github.com/rod-o-walker>, myself also noticed
> this issue.
> i'm working on some major changes, will submit and publish soon.
> here is how i've fixed this issue of the link stylesheet.
>
> document.head.querySelectorAll('link[rel="stylesheet"][type="text/css"]').forEach(node => {
>   this.popoutWindow.document.head.insertAdjacentHTML('beforeend',
>   `<link rel="stylesheet" type="text/css" href="${(node as HTMLLinkElement).href}">`);
> });
>
> what do you think of my solution?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Shemesh/angular-popout-window/pull/3#issuecomment-745136866>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQMBJZB7XVUCAOAP4UINN3SU4NMHANCNFSM4U3NP4FA>
> .
>

Shemesh commented:
it didn't worked for you because of the `[type="text/css"]` in the query selector.
this is working good:
 

```
document.head.querySelectorAll('link[rel="stylesheet"]').forEach(node => {
this.popoutWindow.document.head.insertAdjacentHTML('beforeend',
  `<link rel="stylesheet" type="${(node as HTMLLinkElement).type}" href="${(node as HTMLLinkElement).href}">`);
});
```

Shemesh commented:
@rod-o-walker i have release v2 with all those fixes inside plus few more.
check it out.

rod-o-walker commented:
On closer inspection, I see it is working.

I was surprised this returns back the absolute URL ...

(node as HTMLLinkElement).href

I assumed it would also be the relative one again.

Thx.

______________________________________________


Rod Walker
e-mail: rod.o.walker@gmail.com


On Wed, Dec 16, 2020 at 8:12 AM Shemesh <notifications@github.com> wrote:

> it didn't worked for you because of the [type="text/css"] in the query
> selector.
> this is working good:
>
> document.head.querySelectorAll('link[rel="stylesheet"]').forEach(node => {
> this.popoutWindow.document.head.insertAdjacentHTML('beforeend',
>   `<link rel="stylesheet" type="${(node as HTMLLinkElement).type}" href="${(node as HTMLLinkElement).href}">`);
> });
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Shemesh/angular-popout-window/pull/3#issuecomment-746266269>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQMBJ4VUZVOS3B54PEKFK3SVCW5FANCNFSM4U3NP4FA>
> .
>
